### DISCUSSION TITLE
NotHitBy/HitBy extra parameter

### DISCUSSION THREAD

**Discussion by LasombraDemon** [ikemen-engine/Ikemen-GO#1127] [Feature Request] [answered]
### Is your feature request related to a problem? Please describe.

A new parameter to enable/disable hits, accounting for a given Player ID. This could easily be added as an extra parameter for HitBy/NotHitBy.

For instance, in my football minigame, I want the ball to be able to hit people, and I want people to be able to hit the ball. But I don't want people to be able to hit each other. This puts me into an unsolvable problem that could be addressed via an extra parameter. I'm sure others have run into similar issues, too.

It could also be super useful for multiplayer games.

### Describe the solution you'd like

Syntax could be "NotHitBy: SCA, , ; time:100; PlayerID:Player(1)"; this third parameter could be optional, and this would not break backwards compat.

### Search terms

HitBy NotHitBy

**Comment by potsmugen** [comment]
This would be useful indeed but it wouldn't solve your issue, I think. You'd have to account for an infinite number of helper ID's. Maybe if it also accepted a teamside or player number parameter. And then you still have the issue that NotHitBy doesn't stack, so you'd have to get everything you want inside a single sctrl.

You could also solve it by changing the teamside that each player is on, but that may be too gamebreaking to even consider messing with.

**Reply by LasombraDemon** [reply]
My plan was to use HitBy with the ball's Helper ID on players 1-8, so only the ball can hit people, and yet people can hit the ball.

**Reply by potsmugen** [reply]
Oh, I was forgetting about the reverse logic. That sounds good, yeah.

**Comment by potsmugen** [comment, marked as answer]
Already implemented:
https://github.com/ikemen-engine/Ikemen-GO/pull/1903
